src/openzaak/utils/admin.py

```diff
@@ -118,3 +121,58 @@ class DynamicArrayMixin:
     class Media:
         js = ("js/min/django_better_admin_arrayfield.min.js",)
         css = {"all": ("css/min/django_better_admin_arrayfield.min.css",)}
+
+
+class AuditTrailAdminMixin(object):
+    viewset = None
+
+    def get_viewset(self, request):
+        if not self.viewset:
+            raise NotImplementedError("'viewset' property should be included to the Admin class")
+        return self.viewset(request=request, format_kwarg=None)
+
+    def save_model(self, request, obj, form, change):
+        model = obj.__class__
+        basename = model._meta.object_name.lower()
+
+        viewset = self.get_viewset(request)
+        version, scheme = viewset.determine_version(request, version=api_settings.DEFAULT_VERSION, uuid=obj.uuid)
+        request.version, request.versioning_scheme = version, scheme
+
+        action = 'update' if change else 'create'
```

Comment:
Isn't this an enum somewhere?

Reply:
yeah, `CommonResourceAction`, I'll change it. It's a bit raw now.